extensions/amp-image-slider/1.0/component.js

```diff
@@ -0,0 +1,549 @@
+   * @param {Event} e
+   * @private
+   */
+  const onMouseUp = useCallback(
+    (e) => {
+      e.preventDefault();
+      unlisten(unlistenMouseMove?.current);
+      unlisten(unlistenMouseUp?.current);
+    },
+    [unlisten]
+  );
+  /**
+   * Handler on mouse button down
+   * @param {Event} e
+   * @private
+   */
+  const onMouseDown = useCallback(
+    (e) => {
+      e.preventDefault();
+      pointerMoveX(e.pageX);
+
+      // In case, clear up remnants
+      // This is to prevent right mouse button down when left still down
+      unlisten(unlistenMouseMove?.current);
+      unlisten(unlistenMouseUp?.current);
+
+      unlistenMouseMove.current = listen(win.current, 'mousemove', onMouseMove);
+      unlistenMouseUp.current = listen(win.current, 'mouseup', onMouseUp);
+
+      animateHideHint();
+    },
+    [
+      animateHideHint,
+      pointerMoveX,
+      unlistenMouseMove,
+      unlistenMouseUp,
+      onMouseMove,
+      onMouseUp,
+      unlisten,
+    ]
+  );
+
+  const registerEvents = useCallback(() => {
+    if (isEventRegistered.current) {
+      return;
+    }
+    unlistenMouseDown.current = listen(
+      containerRef.current,
+      'mousedown',
+      onMouseDown
+    );
+    unlistenKeyDown.current = listen(
+      containerRef.current,
+      'keydown',
+      onKeyDown
+    );
+    registerTouchGestures();
+    //this.isEventRegistered = true;
+  }, [
+    registerTouchGestures,
+    unlistenKeyDown,
+    isEventRegistered,
+    onMouseDown,
+    onKeyDown,
+  ]);
+
+  useEffect(() => {
+    /** Common variables */
+    win.current = containerRef.current.ownerDocument.defaultView;
+    doc.current = containerRef.current.ownerDocument;
+    //setIsEdge(Services.platformFor(win).isEdge());
+
+    // TODO (@AnuragVasanwala): Assert Error - Should not contain more than 2 images.
+    // TODO (@AnuragVasanwala): Assert error if images != 2
+    // TODO (@AnuragVasanwala): Assert Error - Should not contain div rather than first and second
+
+    // Notice: hints are attached after amp-img finished loading
+    // buildHint();
+    // checkARIA();
+
+    registerEvents();
+  }, [registerEvents]);
+  useLayoutEffect(() => {
+    /* Do things */
+  }, []);
+  useMemo(() => {
+    /* Do things */
+  }, []);
+
+  /** API Function */
+  useImperativeHandle(
+    ref,
+    () =>
+      /** @type {!BentoImageSliderDef.BentoImageSliderApi} */ ({
+        seekTo: (percent = 50) => {
+          updatePositions(percent);
+        },
+      }),
+    [updatePositions]
+  );
+
+  return (
+    <div
+      ref={containerRef}
+      class={styles.imageSliderContainer}
+      layout
+      size
+      paint
+      tabIndex="0"
+      {...rest}
+    >
+      {/* Masks */}
+      <div ref={leftMaskRef} class={styles.imageSliderLeftMask}>
+        <div ref={leftLabelWrapperRef} class={styles.imageSliderLabelWrapper}>
+          <div ref={leftLabelRef}>{DisplayAs(firstLabelAs)}</div>
+          {DisplayAs(firstImageAs)}
```

Comment:
Let's use `DisplayAs` as a component:
```suggestion
          <div ref={rightLabelRef}>
            <DisplayAs as={secondLabelAs} />
          </div>
          <DisplayAs as={secondImageAs} />
```